cmd/secrets/providers/file.go

```diff
@@ -0,0 +1,100 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// +build secrets
+
+package providers
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	s "github.com/DataDog/datadog-agent/pkg/secrets"
+)
+
+const (
+	maxSecretFileSize = 8192
+)
+
+// FileProvider fetches secrets from the file system.
+type FileProvider struct {
+	rootPath string
+}
+
+// NewFileProvider creates a new FileProvider from the given root directory.
+func NewFileProvider(rootPath string) FileProvider {
+	return FileProvider{rootPath: rootPath}
+}
+
+// Get fetches a secret.
+func (fb *FileProvider) Get(secretID string) s.Secret {
+	return readSecret(filepath.Join(fb.rootPath, secretID))
+}
+
+func readSecret(path string) s.Secret {
```

Comment:
`readSecretFile` is not used anywhere except here. We could remove this wrapper and have `readSecretFile` return a `s.Secret` directly